linera-base/src/identifiers.rs

```diff
@@ -178,6 +174,52 @@ impl<A> Debug for BytecodeId<A> {
     }
 }
 
+impl<A> Serialize for BytecodeId<A> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        if serializer.is_human_readable() {
+            let bytes = bcs::to_bytes(&self.message_id).map_err(serde::ser::Error::custom)?;
+            serializer.serialize_str(&hex::encode(bytes))
+        } else {
+            let mut state = serializer.serialize_struct("BytecodeId", 1)?;
+            state.serialize_field("message_id", &self.message_id)?;
+            state.end()
+        }
+    }
+}
+
+impl<'de, A> Deserialize<'de> for BytecodeId<A> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            let message_id_bytes = hex::decode(s).map_err(serde::de::Error::custom)?;
+            let message_id =
+                bcs::from_bytes(&message_id_bytes).map_err(serde::de::Error::custom)?;
+            Ok(BytecodeId {
+                message_id,
+                _phantom: std::marker::PhantomData,
+            })
+        } else {
+            #[derive(Deserialize)]
+            #[serde(rename = "BytecodeId")]
+            struct Foo {
```

Comment:
Maybe `SerdeBytecodeId`, `SerializableBytecodeId`, or `MyBytecodeId` or something.

I wonder whether it would be better to also derive `Serialize` and use the struct for the `Serialize` implementation above as well. Then any changes only need to be made in one place.

Reply:
Ah yeah, I guess the naming here isn't great haha

Reply:
That one is on me. There are `struct Foo`s in many places probably.